### PR TITLE
Update ray tracing sample to use VK_KHR_ray_tracing

### PR DESCRIPTION
**Pull request by SaschaWillems**
## Description

This PR updates the basic ray tracing sample to use the newly introduced cross-vendor VK_KHR_ray_tracing extension (instead of the vendor-specific VK_NV_ray_tracing).

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [x] My changes build and run on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [x] I have tested the sample on at least one compliant Vulkan implementation
- n/a If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- n/a Any dependent assets have been merged and published in downstream modules

## Notes
- Right now only NVIDIA supports the new extension, so I only tested it on a current NVIDIA GPU (RTX 2060) with current drivers
- The sample requires new functionality in the volk and VMA sub moduls, so I had to update these
- The setup for VMA has been extended with support for buffer device addresses (see the device class)
- The extension is currently beta, which requires ```VK_ENABLE_BETA_EXTENSIONS``` to be enabled. I added this to the CMAKE setup and enabled it by default

### CODE REVIEW

File: samples/extensions/raytracing_basic/raytracing_basic.cpp [hunk 1]
```diff
@@ -100,279 +200,260 @@ void RaytracingBasic::create_storage_image()
...
+		acceleration_device_address_info.sType                 = VK_STRUCTURE_TYPE_ACCELERATION_STRUCTURE_DEVICE_ADDRESS_INFO_KHR;
+		acceleration_device_address_info.accelerationStructure = top_level_acceleration_structure;
+
+		top_level_acceleration_structure_handle = vkGetAccelerationStructureDeviceAddressKHR(device->get_handle(), &acceleration_device_address_info);
+	}
 }
 
 /*
-	Create the Shader Binding Table that binds the programs and top-level acceleration structure
+	Create the Shader Binding Table that holds the ray tracing piepeline's shader group handle
```
Comment (eloj) [inline samples/extensions/raytracing_basic/raytracing_basic.cpp]: Comment typo.

**Review by gary-sweet — CHANGES_REQUESTED** [review]
I see some build warnings:

```
In file included from /build/gsweet/builds/7216_64/libs/vulkan/third_party_demos/KhronosVulkanSamplesExt/src/KhronosVulkanSamples_internal-build/app/samples.h:37:0,
                 from /build/gsweet/vc5_git/thirdparty_vulkan/Vulkan-Samples/app/vulkan_samples.h:23,
                 from /build/gsweet/builds/7216_64/libs/vulkan/third_party_demos/KhronosVulkanSamplesExt/src/KhronosVulkanSamples_internal-build/app/main.cpp:23:
/build/gsweet/vc5_git/thirdparty_vulkan/Vulkan-Samples/app/../samples/extensions/raytracing_basic/raytracing_basic.h:24:0: warning: "VK_KHR_ray_tracing" redefined
 #define VK_KHR_ray_tracing
 
In file included from /build/gsweet/vc5_git/thirdparty_vulkan/Vulkan-Samples/third_party/vulkan/include/vulkan/vulkan.h:88:0,
                 from /build/gsweet/vc5_git/thirdparty_vulkan/Vulkan-Samples/third_party/volk/volk.h:50,
                 from /build/gsweet/vc5_git/thirdparty_vulkan/Vulkan-Samples/framework/common/strings.h:23,
                 from /build/gsweet/vc5_git/thirdparty_vulkan/Vulkan-Samples/framework/common/error.h:24,
                 from /build/gsweet/vc5_git/thirdparty_vulkan/Vulkan-Samples/framework/common/utils.h:20,
                 from /build/gsweet/vc5_git/thirdparty_vulkan/Vulkan-Samples/framework/platform/platform.h:24,
                 from /build/gsweet/builds/7216_64/libs/vulkan/third_party_demos/KhronosVulkanSamplesExt/src/KhronosVulkanSamples_internal-build/app/main.cpp:21:
/build/gsweet/vc5_git/thirdparty_vulkan/Vulkan-Samples/third_party/vulkan/include/vulkan/vulkan_beta.h:85:0: note: this is the location of the previous definition
 #define VK_KHR_ray_tracing 1
```

**Comment by SaschaWillems** [comment]
Thanks for the hint. This was a left-over trying to get volk include the beta functions for ray tracing. Will try to fix this asap.

**Comment by SaschaWillems** [comment]
I have pushed a few updates. The warning reported by  @gary-sweet is now fixed along with another compiler warning due to an implicit type-cast in the sample.

I also removed the VKB CMake option to enable Vulkan beta extension and always enabled it, as disabling would break the build.

**Review by mattruschnv — DISMISSED** [review]

**Review by aurolou — CHANGES_REQUESTED** [review]
Could you please add a warning that this extension is still in beta and will be subject to change? Thanks

**Comment by SaschaWillems** [comment]
> Could you please add a warning that this extension is still in beta and will be subject to change? Thanks

Would a note at the top of the sample's source file be sufficient?

**Comment by ghost** [comment]
> > Could you please add a warning that this extension is still in beta and will be subject to change? Thanks
> 
> Would a note at the top of the sample's source file be sufficient?

Yes, this should be enough in my opinion, please make the update.

**Comment by aurolou** [comment]
> > > Could you please add a warning that this extension is still in beta and will be subject to change? Thanks
> > 
> > 
> > Would a note at the top of the sample's source file be sufficient?
> 
> Yes, this should be enough in my opinion, please make the update.

yes, should be enough. we could also mention that development drivers are needed. it's just to make sure people are not trying to ship ray tracing code as production :)

**Comment by SaschaWillems** [comment]
I have added the following comment to the header and implementation file:

```cpp
/*
 * Important note:
 *
 * The VK_KHR_ray_tracing extension is currently considered a BETA extension.
 *
 * This means that it is not yet production ready and subject to change until it's finalized.
 *
 * In order to use this sample you may also need special developer drivers.
 */
```

Will remove it once the extension is finalized.

I hope this makes it clear that the extension is not yet ready for production.

**Review by wasimabbas-arm — APPROVED** [review]

**Pull request merged**